File: file_processing/pl/ascii_file_sort.pl
#!/usr/bin/perl

###################################################
## slurp lines from file of numbers 
## and sort them lexicographically as strings using
## the default sorting algorithm.
## To run:
## >> perl ascii_file_sort.pl unsorted_numbers.txt
## @author: vladimir kulyukin
###################################################
use strict;
use warnings;

# 1. get the 0th command line argument
my $infile = shift;
# 2. open the file for reading
open(IN, '<', $infile) or die $!;
# 3. read lines, sort, and print
my @lines = <IN>;
print sort { $a cmp $b } @lines;

File: file_processing/py/ascii_file_sort.py
#!/usr/bin/python

#####################################
## reading a file and sorting
## its lines lexicographically.
## @author: vladimir kulyukin
#####################################

import sys

# 1. get the 1st command line argument
file_path = sys.argv[1]
# 2. open the file for reading
infile = open(file_path, 'r')
# 3. read lines, sort, and print
for n in sorted([line for line in infile.readlines()]):
    print n,
